src/lib.rs

```diff
@@ -973,3 +984,35 @@ fn getenv_unwrap(v: &str) -> String {
 fn fail(s: &str) -> ! {
     panic!("\n{}\n\nbuild script failed, must exit now", s)
 }
+
+// Returns the major, minor and patch versions of cmake
+fn cmake_version(executable: &OsStr) -> Option<(u8, u8, u8)> {
+    let cmd = Command::new(executable).arg("--version").output().ok()?;
+    let version = String::from_utf8_lossy(&cmd.stdout);
+    let version: Vec<&str> = version.split_whitespace().collect();
+    let version: Vec<&str> = version[2].split('.').collect();
+    let mut temp = vec![];
+    for i in version.iter() {
+        let curr = i.parse().ok()?;
+        temp.push(curr);
+    }
+    Some((temp[0], temp[1], temp[2]))
```

Comment:
I think you'll want to unroll the loop manually because otherwise there's no guarantee that these indexes are actually present and otherwise this could panic.